### PR TITLE
UX improvements

### PR DESCRIPTION
**Pull request by mrexodia**
This makes the plugin actually usable for me.

### CODE REVIEW

File: plugin.json [hunk 1]
```diff
@@ -1 +1,31 @@
...
+    "name": "Snippet UI Plugin",
+    "type": [
+        "ui"
+    ],
+    "api": [
+        "python2",
+        "python3"
+    ],
+    "description": "Powerful code-editing plugin for writing and managing python code-snippets with syntax highlightingd, hotkey binding and other features",
+    "longdescription": "The snippet editor started as a simple example UI plugin to demonstrate new features available to UI plugins. It has turned into a functionally useful plugin in its own right. The snippet editor allows you to write small bits of code that might not be big enough to warrant the effort of a full plugin but are longer enough that you don't want to retype them every time in the python-console!\n\nAs an added bonus, all snippets are added to the snippets menu and hot-keys can be associated with them as they make use of the action system. All action-system items are also available through the command-palette (CTL/CMD-p).\n\n![](https://github.com/Vector35/snippets/blob/master/media/snippets.gif?raw=true)\n\n.",
```
Comment (galenbwill) [inline plugin.json:12]: "longer enough" should be "long enough"

Comment (psifertex) [reply plugin.json:12]: Yeah, but that's not a new bug == my original text had that wrong. This is just more of a formatting issue. Which actually reminds me that I need to update https://github.com/Vector35/release_helper/blob/master/do_release.py#L40 to have an `indent=` parameter which will prevent this from regressing the next time I use it.

**Comment by psifertex** [comment]
Thanks! Looks like some good changes -- I've also got a branch that implemented snippets as a dock-widget (which fixes the ability to leave it on screen while working with it) but I've kinda held off making that change since the 3.0 UI overhaul is coming in a month or so and it will provide a new sidebar panel which will be perfect.

That said, the non-modal change is a good one in the meantime and I appreciate the improvements.

**Comment by psifertex** [comment]
Ahh, I forgot the other reason I hadn't made this change -- with the non-modal window, the UI context doesn't update the way you'd expect.

So if you for example use `here` or `current_function` and navigate in the UI it won't update. I'll need to fix that or revert this.

**Comment by psifertex** [comment]
Also, just as a heads-up the `dependencies` field wasn't correct in the plugin.json but we're in the middle of debating how we want to change that going forward. The current auto-dependency installation capability relies just on requirements.txt and we were debating removing the dependency from the plugin.json but it is nice when it's generated into the readme as well, so I haven't totally made up my mind. Either way I fixed up the context generation with https://github.com/Vector35/snippets/commit/1d804abe6135df0bc12c67de43f4338c53cd2fff which is the more important change.

**Comment by mrexodia** [comment]
For me it didn’t automatically install the dependencies, but my support ran
out in September 2020 so it might have been fixed already

**Pull request merged**